test/verify/check-pages

```diff
@@ -862,6 +865,84 @@ OnCalendar=daily
         # logging out too fast, some D-Bus services get disconnected
         self.allow_restart_journal_messages()
 
+    @testlib.skipBrowser(".upload_files() doesn't work on Firefox", "firefox")
+    def testUpload(self) -> None:
+        b = self.browser
+        m = self.machine
+
+        self.restore_dir("/home/admin")
+        self.login_and_go("/playground/react-patterns")  # , debugging="upload")
+        files_dir = Path(testlib.TEST_DIR) / "verify" / "files" / "metrics-archives"
+        test_upload_file = str(files_dir / "double_events.zip")
+        dest_file = "/home/admin/double_events.zip"
+
+        filehash = subprocess.check_output(["sha512sum", test_upload_file]).strip().decode().split(' ')[0]
+        filesize = subprocess.check_output(["stat", "--format", "%s", test_upload_file]).strip().decode()
+
+        b.upload_files("#demo-upload input[type='file']", [test_upload_file])
+
+        testlib.wait(lambda: m.execute(f"test -f {dest_file} && echo {dest_file} || echo 'no'").strip() == dest_file)
+        self.assertEqual(m.execute(f"stat --format '%s' {dest_file}").strip(), filesize)
+        self.assertEqual(m.execute(f"sha512sum {dest_file} | awk '{{ print $1 }}'").strip(), filehash)
+        b.wait_visible("#upload-file-btn:not(:disabled)")
+
+        # Various cases of aborted big file upload
+        with tempfile.TemporaryDirectory() as tmpdir:
+            big_file = str(Path(tmpdir) / "bigfile.img")
+            subprocess.check_call(["truncate", "-s", "1G", big_file])
+
+            # Cancelled
+            b.upload_files("#demo-upload input[type='file']", [big_file])
```

Comment:
Please bring back the test of uploading the full file. All this flow control etc. could really do with a stress test that would OOM the target machine if anything goes wrong there.

Reply:
I think for me the issue is that something like:
 - we wanted to check to make sure that we're properly reading the file in chunks, and not into memory all at once
 - if we were reading the file all at once, we'd actually find out about that already at the first block...
 - 1GB is not enough to crash either the browser or the target machine anyway.  we'd need something *much* bigger in that case
 - all of these concerns are kinda no longer relevant imho because the APIs we're using now are super simple to understand — the `ReadableStream` with `byob` is pretty explicit about the fact that it's basically passing my 16k buffer directly into the `read()` syscall... and on the bridge side the handling is also super simple...

So my thinking was that we could skip the entire 1GB (or 10GB or however much) uploading, make the test fast, and then run it always.

Reply:
But still it could be useful no? Stress testing a long upload is not done anywhere else and it was gated behind `DEFAULT_OS` so not every run would eat CPU cycles. (I suppose devel will become slower though)

Reply:
> 1GB is not enough to crash either the browser or the target machine anyway.

It should -- our test VMs have 1400 MB RAM, and a lot of it is taken by the OS and cockpit. So to be really sure you could make it 1.5 GB, but if the bridge did try to slurp the whole file into memory first we really should notice already.

Note that the reason for skipping the test on TF is mostly unrelated to that -- even if it would only take 2 instead of 4 minutes we would still want to skip it there IMHO.

Reply:
I agree about the TF thing.  I misunderstood your intent there.

The original concern wasn't about the bridge, but about the browser.  From what I understand, we'd need to eat 24GB before that became a problem...

If there was a reasonable way to measure browser memory consumption which wasn't likely to break at the next update...

Reply:
Yeah, this obviously only checks the bridge side, not the browser side. But it's also still a nice stress test in general -- wouldn't be the first time that scaling up things discovered bugs..

Reply:
Okay.  It's back, and enabled on `TEST_OS_DEFAULT`.